File: a/encapsulamiento/CajaAhorros.java
package a.encapsulamiento;

public class CajaAhorros {
    public String nombre;
    protected int nroCuenta;
    private double saldo;
    public static void main(String[] args) {
        CajaAhorros caja = new CajaAhorros("Juan Perez", 123456, 1000);
        
        System.out.println("\t----ACCESO AL ATRIBUTO Protegidos---");
        System.out.println("Saldo: " + caja.saldo);
        caja.saldo = 100000;
        System.out.println("\t---ACCESO A LOS METODOS Protegidos---");
        caja.transferencia(300);
        System.out.println("Saldo: " + caja.saldo);
    }

    
    public CajaAhorros(String nombre, int nroCuenta, double saldo) {
        this.nombre = nombre;
        this.nroCuenta = nroCuenta;
        this.saldo = saldo;
    }
    public void mostrarDatos() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Nro. Cuenta: " + nroCuenta);
        System.out.println("Saldo: " + saldo);
    }
    protected double getSaldo(){
        return this.saldo;
    }

    private void transferencia(double monto) {
        this.saldo += monto;
    }

    
    
}

File: a/encapsulamiento/CajaAhorros.py
class CajaAhorros:
    def __init__(self, nombre, nro, saldo):
        self.nombre = nombre
        self._nroCuenta = nro
        self.__saldo = saldo
    
    def mostrarDatos(self):
        print("Nombre: ", self.nombre)
        print("Número de cuenta: ", self._nroCuenta)
        print("Saldo: ", self.__saldo)
        
    def _getSaldo(self):
        return self.__saldo
    
    def __transferencia(self, monto):
        self.__saldo += monto

caja = CajaAhorros("Juan",1112, 100)
print("\t\t-->Acceso a atributos protegidos")
print(caja.__saldo)
caja.__saldo = 100000

print("\t\t-->Acceso a metodos protegidos")
caja.__transferencia(300)
print(caja.__saldo)